c/httpserver.c

```diff
@@ -3800,11 +3808,8 @@ void respondWithUnixFileNotFound(HttpResponse* response, int jsonMode) {
 // Response must ALWAYS be finished on return
 void respondWithJsonError(HttpResponse *response, char *error, int statusCode, char *statusMessage) {
   jsonPrinter *out = respondWithJsonPrinter(response);
-
-  setContentType(response, "text/json");
   setResponseStatus(response,statusCode,statusMessage);
-  addStringHeader(response, "Server", "jdmfws");
-  addStringHeader(response, "Transfer-Encoding", "chunked");
+  setDefaultJSONRESTHeaders(response);
```

Comment:
1. httpserver.c doesn't include dataservice.h thus doesn't have the correct declaration of that new function. This leads to subtle bugs because to the compiler this call will be equivalent to ```int foo(int var)```.
2. httpserver.c, which, I think, shouldn't include dataservice.h will now have to include it. And anyone who wants to use it will have to link with dataservice.c.

Reply:
So, do you recommend putting setDefaultJSONRESTHeaders into httpserver.c & httpserver.h directly? Was trying to avoid bloating this file, but it seems similar functions are already there.

Reply:
For now yes, because otherwise we make the code more intertwined. I'd make some httputils.c in the future.